pkg/sample/random_test.go

```diff
@@ -0,0 +1,867 @@
+				301: 15,
+				302: 15,
+				303: 15,
+				304: 15,
+			},
+		},
+		// Case 004
+		{
+			seq: map[uint]int{
+				400: 20,
+				401: 20,
+				402: 20,
+				403: 20,
+				404: 20,
+			},
+		},
+		// Case 005
+		{
+			seq: map[uint]int{
+				500: 25,
+				501: 25,
+				502: 25,
+				503: 25,
+				504: 25,
+			},
+		},
+		// Case 006
+		{
+			seq: map[uint]int{
+				600: 30,
+				601: 30,
+				602: 30,
+				603: 30,
+				604: 30,
+			},
+		},
+		// Case 007
+		{
+			seq: map[uint]int{
+				995: 50,
+				996: 50,
+				997: 50,
+				998: 50,
+				999: 50,
+			},
+		},
+		// Case 008
+		{
+			seq: map[uint]int{
+				1000:   50,
+				1500:   50,
+				5000:   50,
+				10000:  50,
+				100000: 50,
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("%03d", i), func(t *testing.T) {
+			for k, v := range tc.seq {
+				l := crv(k)
+
+				if l != v {
+					t.Fatalf("%d: expected %#v got %#v", k, v, l)
+				}
+			}
+		})
+	}
+}
+
+func Test_Sample_Random_Crypto(t *testing.T) {
+	testCases := []struct {
+		a uint
+		b uint
+	}{
+		// Case 000
+		{
+			a: 0,
+			b: 0,
+		},
+
+		// Case 001
+		{
+			a: 1,
+			b: 1,
+		},
+
+		// Case 002
+		{
+			a: 2,
+			b: 2,
+		},
+
+		// Case 003
+		{
+			a: 3,
+			b: 3,
+		},
+
+		// Case 004
+		{
+			a: 4,
+			b: 4,
+		},
+
+		// Case 005
+		{
+			a: 5,
+			b: 5,
+		},
+
+		// Case 006
+		{
+			a: 10,
+			b: 10,
+		},
+
+		// Case 007
+		{
+			a: 1,
+			b: 0,
+		},
+		// Case 008
+		{
+			a: 0,
+			b: 1,
+		},
+
+		// Case 009
+		{
+			a: 2,
+			b: 0,
+		},
+		// Case 010
+		{
+			a: 0,
+			b: 2,
+		},
+
+		// Case 011
+		{
+			a: 3,
+			b: 0,
+		},
+		// Case 012
+		{
+			a: 0,
+			b: 3,
+		},
+
+		// Case 013
+		{
+			a: 1,
+			b: 2,
+		},
+		// Case 014
+		{
+			a: 2,
+			b: 1,
+		},
+
+		// Case 015
+		{
+			a: 3,
+			b: 1,
+		},
+		// Case 016
+		{
+			a: 1,
+			b: 3,
+		},
+
+		// Case 017
+		{
+			a: 3,
+			b: 2,
+		},
+		// Case 018
+		{
+			a: 2,
+			b: 3,
+		},
+
+		// Case 019
+		{
+			a: 3,
+			b: 4,
+		},
+		// Case 020
+		{
+			a: 4,
+			b: 3,
+		},
+
+		// Case 021
+		{
+			a: 2,
+			b: 5,
+		},
+		// Case 022
+		{
+			a: 5,
+			b: 2,
+		},
+
+		// Case 023
+		{
+			a: 10,
+			b: 0,
+		},
+		// Case 024
+		{
+			a: 0,
+			b: 10,
+		},
+
+		// Case 025
+		{
+			a: 7,
+			b: 5,
+		},
+		// Case 026
+		{
+			a: 5,
+			b: 7,
+		},
+
+		// Case 027
+		{
+			a: 6,
+			b: 1,
+		},
+		// Case 028
+		{
+			a: 1,
+			b: 6,
+		},
+
+		// Case 029
+		{
+			a: 30,
+			b: 5,
+		},
+		// Case 030
+		{
+			a: 5,
+			b: 30,
+		},
+
+		// Case 031
+		{
+			a: 57,
+			b: 2,
+		},
+		// Case 032
+		{
+			a: 2,
+			b: 57,
+		},
+
+		// Case 033
+		{
+			a: 108,
+			b: 20,
+		},
+		// Case 034
+		{
+			a: 20,
+			b: 108,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("%03d", i), func(t *testing.T) {
+			mpx := map[string]int{}
+			mpy := map[string]int{}
+
+			for i := 0; i < 1000; i++ {
+				var sam *Sample
+				{
+					sam = New(Config{
+						Rea: rand.Reader,
+					})
+				}
```

Comment:
This here is how we can create a new instance of the sampler. The important bit here is to use the `io.Reader` of the `crypto/rand` package. Only that source guarantees enough entropy to shuffle properly.